wizard/src/components/mapping_table_header.tsx

```diff
@@ -56,31 +85,81 @@ export function MappingTableHeader(
           })}
         </tr>
       </thead>
-      <thead>
+      <thead className="column-header-row">
         <tr>
           <th className="row-num">0</th>
-          {props.orderedColumns.map((column, idx) => {
+          {orderedColumns.map((columnIdx) => {
+            const info = props.columnInfo.get(columnIdx);
             return (
               <th
-                key={`heading-${idx}`}
+                key={`heading-${columnIdx}`}
                 className={
-                  props.selectedColumn === idx ? "highlighted-col" : ""
+                  props.selectedColumn === columnIdx ? "highlighted-col" : ""
                 }
                 ref={
-                  props.selectedColumn === idx
+                  props.selectedColumn === columnIdx
                     ? props.highlightedColumnRef
                     : null
                 }
-                onClick={() => props.onColumnSelected(idx)}
+                onClick={() => props.onColumnSelected(columnIdx)}
               >
-                <div>{column.header}</div>
+                {editHeaderState.colIdx === columnIdx ? (
+                  <form
+                    onSubmit={(event) => {
+                      event.preventDefault();
+                      onHeaderUpdated();
+                    }}
+                  >
+                    <Input
+                      className="header-input"
+                      type="text"
+                      multiple={true}
+                      onChange={(e) => {
+                        const val = e.target.value;
+                        setEditHeaderState({ ...editHeaderState, header: val });
+                      }}
+                      value={editHeaderState.header}
+                      onBlur={() => onHeaderUpdated()}
+                      autoFocus={true}
+                    />
+                  </form>
+                ) : (
+                  <div>
+                    <span>{info.column.header}</span>
+                    <i
+                      onClick={() =>
+                        setEditHeaderState({
+                          colIdx: columnIdx,
+                          header: info.column.header,
+                        })
+                      }
+                      className="material-icons-outlined"
+                    >
+                      edit
+                    </i>
+                  </div>
+                )}
               </th>
             );
           })}
         </tr>
       </thead>
     </>
   );
+
+  function onHeaderUpdated(): void {
+    const info = props.columnInfo.get(editHeaderState.colIdx);
+    const updatedColumn = {
+      ...info.column,
+      id: editHeaderState.header,
+      header: editHeaderState.header,
+    };
+    props.onColumnUpdated(editHeaderState.colIdx, {
+      ...info,
+      column: updatedColumn,
+    });
```

Comment:
To confirm, if after user edit there are duplicates how do we handle it?

Reply:
Currently, this would just end up being a duplicate column header. Some other ideas to handle this: 

a. automatically attach the column index to the header name if it's a duplicate
b. validate that all headers are unique and show an error message if there's a duplicate